compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -2590,6 +2597,28 @@ class Typer extends Namer
           }
         case _ =>
       }
+      // try an extension method in scope
+      pt match {
+        case SelectionProto(name, mbrType, _, _) =>
+          def tryExtension(implicit ctx: Context): Tree =
+            try {
+              findRef(name, WildcardType, ExtensionMethod, tree.pos) match {
+                case ref: TermRef =>
+                  extMethodApply(untpd.ref(ref).withPos(tree.pos), tree, mbrType)
+                case _ => EmptyTree
+              }
+            }
+            catch {
+              case ex: TypeError => errorTree(tree, ex.toMessage, tree.pos)
+            }
+          val nestedCtx = ctx.fresh.setNewTyperState()
+          val app = tryExtension(nestedCtx)
+          if (!app.isEmpty && !nestedCtx.reporter.hasErrors) {
+            nestedCtx.typerState.commit()
+            return Applications.ExtMethodApply(app).withType(app.tpe)
```

Comment:
Looks like `ExtMethodApply` is always created around the result of a call to `extMethodApply`, can `extMethodApply` be changed to directly return an `ExtMethodApply` ?

Reply:
I don't see it being created when it comes from typedImplicit